### PR TITLE
[Do Not Merge] Catch AuthorizationException

### PR DESCRIPTION
**Pull request by nanoscott**
We don't want the authorization exception propogated to the FrontEnd, so I'm catching it and returning null.

### CODE REVIEW

File: src/foam/nanos/auth/AuthorizationDAO.js [hunk 1]
```diff
@@ -91,8 +91,14 @@ foam.CLASS({
...
 
     FObject obj = super.find_(x, id);
-    if ( obj != null ) getAuthorizer().authorizeOnRead(x, obj);
-    return obj;
+    try 
+    {
+      if ( obj != null ) 
+        getAuthorizer().authorizeOnRead(x, obj);
+      return obj;
+    } catch (Exception e) {
```
Comment (TW80000) [inline src/foam/nanos/auth/AuthorizationDAO.js]:
```suggestion
    } catch (AuthorizationException e) {
```

File: src/foam/nanos/auth/AuthorizationDAO.js [hunk 2]
```diff
@@ -91,8 +91,14 @@ foam.CLASS({
     if ( getAuthorizer().checkGlobalRead(x) ) return super.find_(x, id);
 
     FObject obj = super.find_(x, id);
-    if ( obj != null ) getAuthorizer().authorizeOnRead(x, obj);
-    return obj;
+    try 
+    {
```
Comment (TW80000) [inline src/foam/nanos/auth/AuthorizationDAO.js]: This should at the end of the line above.

**Review by TW80000 — APPROVED** [review]
I agree, this should have been the behaviour all along. I have a feeling this PR is likely to break some existing code though, so we should be careful when merging.

**Comment by nanoscott** [comment]
I'm getting 2 new unit test failures, so marking as DNM for now.

image

**Review by jlhughes — APPROVED** [review]

**Pull request merged**